packages/adblocker/src/engine/engine.ts

```diff
@@ -294,6 +309,9 @@ export default class FilterEngine extends EventEmitter<
 
   public lists: Map<string, string>;
 
+  public env: Env;
```

Comment:
I don't think we should keep references to `env` in different places (or at all). If we need to keep a reference to `env` then let's figure out what's the best place for it.

Reply:
Yup, `PreprocessorBucket` or `FiltersEngine`. If we need to serialize it, I'd like to prefer `PreprocessorBucket` over `FiltersEngine` to make serialization step simple.

Reply:
Just a note: The `FiltersEngine.env` is now removed but still `FiltersEngine.updateEnv` exists.